src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.js

```diff
@@ -0,0 +1,146 @@
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import environment from 'platform/utilities/environment';
+import { apiRequest } from 'platform/utilities/api';
+
+class CernerCallToAction extends Component {
+  static propTypes = {
+    type: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+    // From mapStateToProps.
+    facilities: PropTypes.arrayOf(
+      PropTypes.shape({
+        facilityId: PropTypes.string.isRequired,
+        isCerner: PropTypes.bool.isRequired,
+      }).isRequired,
+    ).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: '',
+      facilities: [],
+      fetching: true,
+    };
+  }
+
+  componentDidMount() {
+    const { facilities } = this.props;
+
+    // Escape early if there are no facilities.
+    if (isEmpty(facilities)) {
+      return;
+    }
+
+    // Derive the cerner facilities.
+    const cernerFacilities = facilities.filter(facility => facility.isCerner);
+
+    // Escape early if there are no cerner facilities.
+    if (isEmpty(cernerFacilities)) {
+      // WARNING: Add sentry logging here if there are no cerner facilities found, this should never happen as the component only renders when there ARE cerner facilities.
+      return;
+    }
+
+    // Derive the list of facility IDs.
+    const cernerFacilityIDs = cernerFacilities.map(
+      facility => `vha_${facility.facilityId}`,
+    );
+
+    // Fetch cerner facilities.
+    this.fetchFacilities(cernerFacilityIDs);
+  }
+
+  fetchFacilities = async facilityIDs => {
+    // Show loading state.
+    this.setState({ fetching: true });
+
+    try {
+      // Fetch facilities and store them in state.
+      const response = await apiRequest(
+        `${environment.API_URL}/v1/facilities/va?ids=${facilityIDs.join(',')}`,
+      );
+      this.setState({ facilities: response?.data, fetching: false });
+
+      // Log any API errors.
+    } catch (error) {
+      this.setState({ error: error.message, fetching: false });
+    }
+  };
+
+  render() {
+    const { text, type } = this.props;
+    const { error, fetching, facilities } = this.state;
+
+    // Escape early if we are fetching.
+    if (fetching) {
+      return <LoadingIndicator message="Loading your information..." />;
+    }
+
+    // Escape early if there was an error fetching the Cerner facilities.
+    if (error || isEmpty(facilities)) {
+      return (
+        <AlertBox
+          headline="Something went wrong"
+          content="We’re sorry. Something went wrong on our end. Please try again later."
+          status="error"
+        />
+      );
+    }
+
+    // Derive the Cerner facility names.
+    const facilityNames = facilities.map(
+      facility => facility?.attributes?.name || 'unknown facility name',
+    );
+    const joinedFacilityNames =
+      facilityNames.join(', ') || 'unknown cerner facility(s)';
```

Comment:
This is a little confusing with the default value in `'unknown facility name'` and the next one in `'unknown cerner facility(s)'`. I suppose that condition shouldn't really be run into anyway, but wondering if there's a way to make it clearer. I also completely forgot that empty arrays evaluate to false lol.

Reply:
Oh empty arrays evaluate to true, you were right 🙂 but `.join` makes it a string, and an empty string is falsey... there's probably a better way to handle this indeed, hmm...